controllers/provisioners/eks/update.go

```diff
@@ -206,6 +227,43 @@ func (ctx *EksInstanceGroupContext) ScalingGroupUpdateNeeded() bool {
 	return false
 }
 
+func (ctx *EksInstanceGroupContext) SuspendProcessesUpdateNeeded() (bool, []string, []string) {
+	var (
+		instanceGroup         = ctx.GetInstanceGroup()
+		configuration         = instanceGroup.GetEKSConfiguration()
+		state                 = ctx.GetDiscoveredState()
+		scalingGroup          = state.GetScalingGroup()
+		specSuspendProcesses  = configuration.GetSuspendProcesses()
+		groupSuspendProcesses []string
+	)
+
+	for _, element := range scalingGroup.SuspendedProcesses {
+		groupSuspendProcesses = append(groupSuspendProcesses, *element.ProcessName)
+	}
+
+	if !common.StringSliceEqualFold(specSuspendProcesses, groupSuspendProcesses) {
+		return true, difference(specSuspendProcesses, groupSuspendProcesses), difference(groupSuspendProcesses, specSuspendProcesses)
+	}
+
+	return false, nil, nil
+}
+
+// Set Difference: A - B
+func difference(a, b []string) (diff []string) {
+	m := make(map[string]bool)
+
+	for _, item := range b {
+		m[item] = true
+	}
+
+	for _, item := range a {
+		if _, ok := m[item]; !ok {
+			diff = append(diff, item)
+		}
+	}
+	return
```

Comment:
should this return a diff?

Can you also move this method to common package?

Reply:
Hi,
I have moved it to the common package. I had previously used a [naked return](https://www.ardanlabs.com/blog/2013/10/functions-and-naked-returns-in-go.html)
Have refactored it to use the conventional return style.

Thank you.